monai/networks/blocks/selfattention.py

```diff
@@ -86,9 +90,17 @@ def __init__(
         self.hidden_input_size = hidden_input_size if hidden_input_size else hidden_size
         self.out_proj = nn.Linear(self.inner_dim, self.hidden_input_size)
 
-        self.qkv = nn.Linear(self.hidden_input_size, self.inner_dim * 3, bias=qkv_bias)
-        self.input_rearrange = Rearrange("b h (qkv l d) -> qkv b l h d", qkv=3, l=num_heads)
-        self.out_rearrange = Rearrange("b h l d -> b l (h d)")
+        if use_combined_linear:
+            self.qkv = nn.Linear(self.hidden_input_size, self.inner_dim * 3, bias=qkv_bias)
+            self.to_q = self.to_k = self.to_v = nn.Identity()  # add to enable torchscript
+            self.input_rearrange = Rearrange("b h (qkv l d) -> qkv b l h d", qkv=3, l=num_heads)
+        else:
+            self.to_q = nn.Linear(self.hidden_input_size, self.inner_dim, bias=qkv_bias)
+            self.to_k = nn.Linear(self.hidden_input_size, self.inner_dim, bias=qkv_bias)
+            self.to_v = nn.Linear(self.hidden_input_size, self.inner_dim, bias=qkv_bias)
+            self.qkv = nn.Identity()  # add to enable torchscript
+            self.input_rearrange = Rearrange("b h (l d) -> b l h d", l=num_heads)
+        self.out_rearrange = Rearrange("b l h d -> b h (l d)")
```

Comment:
If we don't have a compelling reason for this to be an option, we can just use the 3 linear layers without this being a choice. We haven't released 1.4 yet so nothing here should be dependent yet.

Reply:
The combined way was introduced since SABlock added. I've mentioned [here](https://github.com/Project-MONAI/MONAI/issues/7992#issuecomment-2270802435) that there will be a big gap between the two, and maybe we should make this an option, otherwise for the user, the behavior will change without any notice. What do you think?
https://github.com/Project-MONAI/MONAI/commit/d41af7e498677ed813aeb39bc541bc36f72e86cb#diff-bb1ecca81545a8d2ce544bd6115ddb48e733ad85ea4fdb1edf245ce73c0db6b2

Also cc @Nic-Ma for vis.

Reply:
Yes, it's true the behaviour is different but I don't know if there's any users yet who would be impacted by making it 3 linears.

Reply:
In an earlier conversation with @KumoLiu , it seems that the trained MAISI model is already using the combined way.

Reply:
@mingxin-zheng you had me worried for a moment! I would propose we just use the 3 linears and not bother with an option. The combined way seems to have been an optimisation that isn't necessary.

Reply:
I'm not sure about this one, add it as an option or directly use three linears.
And for your reference.
https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer.py#L117

Open for this one, want to hear more opinion from other person.
cc @Nic-Ma @mingxin-zheng @heyufan1995 @yiheng-wang-nv @dongyang0122 @guopengf

Reply:
It seems like treating it as an option is more flexible.

Reply:
Since `use_combined_layer=True` is for backward compatibility, so I did some search in tag `1.3.1` for the usage of `SABlock`. 

It turns out that `TransformerBlock` is using it, and `TransformerBlock` is used utilized by `ViT`(`UNETR` depends on it) and `ViTAutoEnc`

https://github.com/Project-MONAI/MONAI/blob/96bfda00c6bd290297f5e3514ea227c6be4d08b4/monai/networks/blocks/transformerblock.py#L56

Removing the combined layer, in my observation, will introduce weight loading issue if users are using the blocks and models above, although it is not a fundamental change (user can combine the 3 inner_dim channels manually).

Reply:
I think we've concluded that we leave the option in, I'm good with that.

Reply:
@ericspod Thanks, I also agree to keep the option.